### ISSUE TITLE
Observable decorator error check for setters

### ISSUE THREAD

**Issue by jameslk** [mweststrate/mobservable#121]
Just based on a cursory look over some code, given this line:
https://github.com/mweststrate/mobservable/blob/1.2.5/src/core.ts#L346

```
 const descriptor:PropertyDescriptor = {};
```

it seems like this line will never be true:

https://github.com/mweststrate/mobservable/blob/1.2.5/src/core.ts#L365

```
        if (descriptor.set)
            throw new Error(`@observable properties cannot have a setter (in member: '${key}').`);
```
since `descriptor` is assigned a blank object.

I'm not sure if that's a bug, but it looks like a change was made to `descriptor` in another commit, where it used to be assigned some initial values.

**Comment by mweststrate** [comment]
Fixed and added test. Will be released as part of 2.0. Thanks for spotting!

**Comment by jameslk** [comment]
No problem. I wasn't sure if I had caught a bug or was just reading the code wrong, but I'm glad I could help :)
